master/internal/api_experiment.go

```diff
@@ -205,28 +205,40 @@ func (a *apiServer) GetSearcherEvents(ctx context.Context, req *apiv1.GetSearche
 		return nil, err
 	}
 	if !isActiveExperimentState(exp.State) {
-		event := experimentv1.SearcherEvent_ExperimentInactive{
-			ExperimentInactive: &experimentv1.ExperimentInactive{
-				ExperimentState: exp.State,
-			},
-		}
-		searcherEvent := experimentv1.SearcherEvent{
-			Id:    int32(-1),
-			Event: &event,
-		}
-		events := []*experimentv1.SearcherEvent{&searcherEvent}
-		resp = &apiv1.GetSearcherEventsResponse{
-			SearcherEvents: events,
-		}
-		return resp, nil
+		return &apiv1.GetSearcherEventsResponse{
+			SearcherEvents: []*experimentv1.SearcherEvent{{
+				Id: -1,
+				Event: &experimentv1.SearcherEvent_ExperimentInactive{
+					ExperimentInactive: &experimentv1.ExperimentInactive{
+						ExperimentState: exp.State,
+					},
+				},
+			}},
+		}, nil
 	}
 
 	addr := experimentsAddr.Child(req.ExperimentId)
-	switch err = a.ask(addr, req, &resp); {
-	case err != nil:
-		return nil, status.Errorf(codes.Internal, "failed to get events from actor %v", err)
-	default:
-		return resp, nil
+	var w searcher.EventsWatcher
+	err = a.ask(addr, req, &w)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal,
+			"failed to get events from actor: long polling %v", err)
+	}
+
+	defer a.ask(addr, UnwatchEvents{w.ID}, &w) //nolint
```

Comment:
This should be fixable, so reminder to come back here.

Reply:
In preemption long polling, we've just used a tell here instead.

Reply:
Using tell here instead.